QueueImplementation/VM/CustomCollectionViewModel.cs

```diff
@@ -0,0 +1,363 @@
+		private int? elementToProcess;
+
+		/// <summary>
+		/// Holds the is choice possible.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		private bool isChoisePossible = true;
+
+		/// <summary>
+		/// Holds the message to display.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		private string message;
+
+		/// <summary>
+		/// Holds the remove value command.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		private RelayCommand remove;
+
+		/// <summary>
+		/// Holds the show value command.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		private RelayCommand show;
+
+		/// <summary>
+		/// Gets the add command.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		/// <value>The add command.</value>
+		public RelayCommand Add
+		{
+			get
+			{
+				if (this.add is null)
+					this.add = new RelayCommand(this.AddNumber, this.CanAddNumber);
+
+				return this.add;
+			}
+		}
+
+		/// <summary>
+		/// Adds the number to the collection.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		/// <param name="obj">The object.</param>
+		private void AddNumber(object obj)
+		{
+			if (this.Collection is null)
+				return;
+
+			int previousCount = this.Collection.Count();
+			this.RestoreCurrentElement();
+			int newValue = RandomNumberGenerator.GetValue();
+			this.Collection.Add(newValue);
+			this.Message = $"The item {newValue} was added.";
+
+			this.RaisePropertyChanged(() => this.AllNumbers);
+
+			if (previousCount == 0)
+				this.RestoreRemoveAndShowCommands();
+		}
+
+		/// <summary>
+		/// Gets the collection of all numbers.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		/// <value>The collection of all numbers.</value>
+		public IEnumerable<int> AllNumbers
+		{
+			get
+			{
+				if (this.allNumbers != null && this.Collection != null && this.allNumbers.SequenceEqual(this.Collection))
+					return this.allNumbers;
+
+				if (this.Collection != null)
+				this.allNumbers = new ObservableCollection<int>(this.Collection);
+				else
+					this.allNumbers = new ObservableCollection<int>();
+
+				return this.allNumbers;
+			}
+		}
+
+		/// <summary>
+		/// Gets the possibility of adding new number.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		/// <param name="obj">The object.</param>
+		/// <returns><c>true</c> if adding new number is possible; otherwise, <c>false</c></returns>
+		private bool CanAddNumber(object obj) => this.AllNumbers != null;
+
+		/// <summary>
+		/// Gets the possibility of getting the number.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		/// <param name="obj">The object.</param>
+		/// <returns><c>true</c> if getting number is possible; otherwise, <c>false</c></returns>
+		private bool CanGetNumber(object obj) => this.AllNumbers.Count() > 0;
+
+		/// <summary>
+		/// Gets the collection.
+		/// </summary>
+		/// <owner>Anton Petrenko</owner>
+		/// <value>The collection.</value>
+		public ICustomCollection<int> Collection
```

Comment:
Don't see usages of this collection on the UI.
To be honest it is not clear for me the idea with two collections - Collection and AllNumbers.

Reply:
Now the collection with all numbers is only one as it should be.